Admin: add new About page.

eliorivero opened the pull request:
For now it's just a skeleton with the wrappers needed and without the nav buttons. Needs content and styles.

Details in p1HpG7-6v8-p2

<img width="1137" alt="Captura de pantalla 2019-03-29 a la(s) 11 27 45" src="https://user-images.githubusercontent.com/1041600/55239626-d2722e80-5215-11e9-9a57-907b1cee38c5.png">

<!--- Provide a general summary of your changes in the Title above -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Introduce new About page as a submenu of the **Jetpack** main menu item.
* Introduce new parameter `show-nav` for `Jetpack_Admin_Page::wrap_ui` so buttons can be hidden when passing `false`.
* Introduce new class `Jetpack_About_Page` to implement the **About** page

#### Testing instructions:
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to the Jetpack menu and click the About page. It should display the page properly formatted as the mockups
* Interactive elements should work
* Tracks should be recorded

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->

* Introduce new About page.

eliorivero committed:
Admin: add new About page. For now it's just an skeleton. Needs content and styles.

Introduce new parameter show-nav for Jetpack_Admin_Page::wrap_ui so buttons can be hidden when passing false.
Introduce new class Jetpack_About_Page to implement the About page as a submenu of the Jetpack main menu item.

crunnells commented:
For simplicity I'm going to build out the design elements in this branch/PR as well.

jeffgolenski commented:
@crunnells Just as a sidenote, to avoid any sort of possible confusion, the sidebar link should read "About Jetpack", not just "About" - I don't want any folks possibly thinking that means about WordPress as a whole. cc @joanrho

crunnells commented:
@jeffgolenski Got it, I've changed the link name. I'll be updating this PR later today.

crunnells committed:
Update to About page content, functionality, and styling. Close, but not finished.

crunnells committed:
Update to squash some PHP warnings

crunnells committed:
Added placeholder Gravatars

crunnells committed:
Fix for PHP warning

crunnells committed:
Desktop styling cleanup

crunnells committed:
Added responsive styling and some changes to class names to prevent having to override JP/core css

crunnells commented:
@jeffgolenski With the most recent commit, the styling should be done. Added some tweaks for responsive styling too.

@eliorivero One thing that I need help doing is to pull a8c data (# of a11ns, countries, languages) and a11n Gravatar URIs from automattic.com/about. I was thinking we'd need to add a URI endpoint on automattic.com that spits out some JSON variables so we can insert them into the page. Your thoughts?

jeffgolenski commented:
@crunnells We're getting close. super excited here. Just a couple of small details and we'll be good, I think.

1) vertically align text with arrow of "back" link

![Screen Shot 2019-04-08 at 4 32 18 PM](https://user-images.githubusercontent.com/214813/55755032-03274480-5a1c-11e9-8228-4bc316e19f4b.png)

2) can we change "meet the team" link to say "Meat the automattic team" - just to balance the visual width of the gallery with the text.

3) **Important:** I noticed that when I click "more details" and then the blue "install" button in the dialog, nothing happens.

![Screen Shot 2019-04-08 at 4 43 15 PM](https://user-images.githubusercontent.com/214813/55755629-8ac18300-5a1d-11e9-81dc-1f1e5698e25f.png)

4) Can we move "about jetpack" to sit below "site stats"

![Screen Shot 2019-04-08 at 4 44 40 PM](https://user-images.githubusercontent.com/214813/55755674-a6c52480-5a1d-11e9-92b8-2a1a93728748.png)

5) Can the footer link "take a look at our WordPress.org profile" link to https://profiles.wordpress.org/automattic/#content-plugins instead of https://profiles.wordpress.org/automattic/  _(it's more relevant than the list of "activity" that initially displays)_

6) Can we make the horizontal and vertical padding / margin between each of these cells, match?

![Screen Shot 2019-04-08 at 4 46 55 PM](https://user-images.githubusercontent.com/214813/55755776-fb689f80-5a1d-11e9-8568-2bae52c47adc.png)


7) **Most importantly**. Do we have tracks set up to track all the links on this page? I want to make sure we're tracking clicks on all the about page links and the install / activate links

Everything else looks great on all devices and functions well. Thanks Chris!

crunnells committed:
Update styling and added Tracks events

crunnells commented:
@jeffgolenski With the most recent commit, the styling should be done. Added some tweaks for responsive styling too.

@eliorivero Two things that I need help with is to pull a8c data (# of a11ns, countries, languages) and a11n Gravatar URIs from automattic.com/about. I was thinking we'd need to add a URI endpoint on automattic.com that spits out some JSON variables so we can insert them into the page.
Second, I tried adding the `jptracks` class to the modal links in the plugin cards, but it broke the modal JS. I think we might need to take another approach here.  Your thoughts?

eliorivero commented:
@crunnells I'll try to get you something for the avatars 👍 Let's talk on Slack about that `jptracks`

jeffgolenski commented:
@crunnells Looking ACE! Just one thing I noticed and I think we'll be good on the design front. Can you just adjust the max-width / margin of this container div so it lines up with the jetpack logo? Should be a super quick change.

![Screen Shot 2019-04-11 at 6 53 13 PM](https://user-images.githubusercontent.com/214813/55998308-c06ba380-5c8b-11e9-8051-1b5ec8f4ff4c.png)

![Screen Shot 2019-04-11 at 6 53 18 PM](https://user-images.githubusercontent.com/214813/55998368-edb85180-5c8b-11e9-925f-d7f62c543379.png)

crunnells committed:
Removed .page-content container (which enforced a max-width) and tweaked the back link padding to visually center the link

crunnells commented:
@jeffgolenski I removed the container in 73f5b3c638994e0c8c655d0465c7682362bb81f4 so the content fills the space, and made some tweaks to the Back link container to make sure the whole thing was vertically centered.

eliorivero committed:
Don't prevent default when calling tracks event if the link has the thickbox class to allow it to open the modal

eliorivero committed:
If jptracked link has thickbox on it, don't follow the link. Add jptracks to plugin title and More details link in About page.

eliorivero commented:
@jeffgolenski @crunnells wrote an endpoint to get 8 random Gravatar hashes from the public a12s in D27086-code. It's pending review.

crunnells commented:
Thanks @eliorivero, I left a comment on your endpoint code. The only other request @jeffgolenski had is:

> Can we move "about jetpack" to sit below "site stats"

![Screenshot](https://user-images.githubusercontent.com/214813/55755674-a6c52480-5a1d-11e9-92b8-2a1a93728748.png)

eliorivero committed:
Reorder menu items so the About page is below Site Stats, if the latter exists

eliorivero commented:
Done

<img width="220" alt="Captura de Pantalla 2019-04-18 a la(s) 13 29 09" src="https://user-images.githubusercontent.com/1041600/56376380-08378100-61de-11e9-9875-aa6f7ba96f3e.png">

beaulebens commented on _inc/lib/admin-pages/class.jetpack-about-page.php:
```diff
@@ -0,0 +1,557 @@
+
+			// should probably add some error checking here too
+			$api = plugins_api( 'plugin_information', $args );
+			$plugins[] = $api;
```
Can we also custom sort this list, or randomize it or something? It felt really weird when I viewed this to see Crowdsignal (low rating, no icon) show up, vs something like WooCommerce (super popular, well-branded, we'd like to encourage it being installed, etc).

beaulebens replied:
Urrr oops, I guess this applies actually to the list of plugins we load up above, on line 207.

jeffgolenski replied:
@beaulebens That was an overlook on my part. I had woo in the original mockups, I'd like to have the plugins listed by order of popularity. Woo, Akismet (usually pre-installed), WP Super Cache, VP, CrowdSignal. What do you think?

beaulebens replied:
Let's get WP Super Cache, WooCommerce, WP Job Manager, Co Authors Plus in there. Would rather not continue promoting VaultPress separately.

jeffgolenski replied:
@crunnells Can you work on getting the plugins listed in the order @beaulebens specified above?

crunnells replied:
Done!

eliorivero committed:
Fetch Gravatar hashes for public A12s from wpcom and display them as a list.

eliorivero commented:
@crunnells I've added the portion of code to fetch the Gravatar hashes for public A12s from wpcom and display them as a list.

kraftbj commented:
I'm working on reviewing this today. Since the admin-page is a new file, I'm going to bring it up to full PHPCS compliance while reviewing.

kraftbj committed:
First pass of PHPCS

kraftbj commented:
I pushed up a number of PHPCS updates, but haven't added it to the whitelist yet. There are a lot of issues with no domain on translations, I presume since we're using Core strings. We need to either own the translation or add exceptions for each. The downside for exempting is we need to ensure that we change if/when Core changes (unit tests?).

Likewise, a lot of escaping errors that we need to fix or exempt.
```
FILE: /Users/kraft/code/jetpack/_inc/lib/admin-pages/class-jetpack-about-page.php
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 42 ERRORS AFFECTING 31 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 335 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 338 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 343 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 357 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 360 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 370 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 373 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 375 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 386 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 388 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 402 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 430 | ERROR | All output should be run through an escaping function (like esc_html_e() or esc_attr_e()), found '_e'. (WordPress.Security.EscapeOutput.UnsafePrintingFunction)
 430 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 434 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 434 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 435 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'self_admin_url'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 442 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 442 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 443 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'self_admin_url'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 448 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 448 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 454 | ERROR | All output should be run through an escaping function (like esc_html_e() or esc_attr_e()), found '_e'. (WordPress.Security.EscapeOutput.UnsafePrintingFunction)
 454 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 458 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 458 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 459 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'self_admin_url'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 463 | ERROR | All output should be run through an escaping function (like esc_html_e() or esc_attr_e()), found '_e'. (WordPress.Security.EscapeOutput.UnsafePrintingFunction)
 463 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 467 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 467 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 481 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$title'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 487 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$description'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 491 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$details_link'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 491 | ERROR | All output should be run through an escaping function (like esc_html_e() or esc_attr_e()), found '_e'. (WordPress.Security.EscapeOutput.UnsafePrintingFunction)
 506 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'number_format_i18n'.
     |       | (WordPress.Security.EscapeOutput.OutputNotEscaped)
 513 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 517 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 522 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 522 | ERROR | Missing $domain arg. (WordPress.WP.I18n.MissingArgDomain)
 522 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$active_installs_text'.
     |       | (WordPress.Security.EscapeOutput.OutputNotEscaped)
 530 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$action_links'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
 586 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$output'. (WordPress.Security.EscapeOutput.OutputNotEscaped)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

kraftbj committed:
PHPCS Run 2

kraftbj reviewed (dismissed):
A number of phpcs issues in the new file that I'd like resolved and added to the phpcs whitelist.

kraftbj committed:
Merge branch 'master' into add/about-page

crunnells committed:
Update plugins shown on the About page

crunnells committed:
Added Jetpack domain to translation strings, escaped some outputs

eliorivero committed:
Fix lines that needed escaping. Add About page file to PHPCS whitelist.

jeherve commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,599 @@
+					set_transient( 'a12s_hashes', $hashes, DAY_IN_SECONDS );
+				} else {
+					$hashes = array(
+						'https://2.gravatar.com/avatar/5ef318426c941cbef6db5342c1356231',
```
What do you think about adding a bit of diversity in that default list? Right now it's 16 white males. While that list represented a8c in its first few years, I think we have a more diverse group of people now.

eliorivero replied:
Nice. Done.

jeherve commented on _inc/lib/admin-pages/class.jetpack-admin-page.php:
```diff
@@ -52,11 +53,11 @@ function add_actions() {
-		add_action( "admin_print_scripts-$hook", array( $this, 'admin_scripts'   ) );
+		add_action( "load-$hook", array( $this, 'admin_help' ) );
+		add_action( "load-$hook", array( $this, 'admin_page_load' ) );
+		add_action( "admin_head-$hook", array( $this, 'admin_head' ) );
```
Is this used anywhere?

eliorivero replied:
Good catch! The only reference I found was in this old commit 
https://github.com/Automattic/jetpack/commit/7436091201880f5163d1b5c1deba5f18e7f9411c#diff-46a389aadbd83b4fa94c516047c94018R36 five years ago.
It's used to fire the action `'jetpack_module_configuration_head_' . $_GET['configure']` 
I couldn't find that action in our current code so I'll remove this.

jeherve commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,599 @@
+							?>
+						</p>
+						<p>
+							<a href="https://automattic.com/jobs" target="_blank"  class="jptracks" data-jptracks-name="jetpack_about_work_with_us">
```
Should we use the final URL for that page instead?

```suggestion
							<a href="https://automattic.com/work-with-us/" target="_blank"  class="jptracks" data-jptracks-name="jetpack_about_work_with_us">
```

eliorivero replied:
Done

jeherve commented on _inc/client/components/about-page/style.scss:
```diff
@@ -0,0 +1,180 @@
+		}
+
+		.plugin-icon {
+			height: 84px;
```
This seems to cause the plugin icons to appear smaller than on the Plugins > Add New screen:

![image](https://user-images.githubusercontent.com/426388/56652783-78398180-668c-11e9-8737-ee72e0da2cf5.png)
![image](https://user-images.githubusercontent.com/426388/56652802-7ff92600-668c-11e9-878c-0cc0bd06f3f8.png)

crunnells replied:
This was based on the mockup, but I'd be happy to adjust larger. cc: @jeffgolenski

jeffgolenski replied:
Good eye Jeremy. Doesn't need to be consistent @crunnells - I think I did that in case the plugins had a lot of descriptive text, they would have more room for content.

crunnells replied:
Fixed in 6b18501; plugin icons are now same size as core (128x128px).

jeherve commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,599 @@
+				}
+			}
+
+			$details_link = self_admin_url(
```
This does not quite work on Multisite installs; if you click on the "More Details" link, the modal pops up but you won't see the option to install that plugin; you are in fact redirected to `wp-admin/network/plugin-install.php` with no parameters inside that modal.

eliorivero replied:
Leaving to @crunnells so he decides what should we do.

crunnells replied:
@eliorivero @jeherve Is there a way we can link Multisite to the plugin install page with the slug pre-populated in the search field? I copied the code that drives the plugin page here, but didn't think multisite would be different. I'd rather not spend a bunch of time re-inventing the wheel here.

jeffgolenski replied:
Let me know what you gents can figure out here. I don't personally see this as a blocker. Our multisite installs are fairly low, aren't they? If all else fails we can just send all multisite installs to the plugins page in wp-admin when any of the links in the about page are clicked.

crunnells replied:
Not sure if this helps with context at all, but I basically copied the `[WP_Plugin_Install_List_Table::display_rows()](https://core.trac.wordpress.org/browser/tags/5.1.1/src/wp-admin/includes/class-wp-plugin-install-list-table.php#L437)` method to display the plugin info here.

jeherve replied:
> Is there a way we can link Multisite to the plugin install page with the slug pre-populated in the search field?

On multisite, the plugin search URL looks like this:

`https://mainnetworkurl.com/wp-admin/network/plugin-install.php?s=wp-super-cache&tab=search&type=term`

eliorivero replied:
Fixed

jeherve reviewed (dismissed):
I left a few comments below.

For some reason I got a 503 error for one of the Gravatars:

![image](https://user-images.githubusercontent.com/426388/56652574-fba6a300-668b-11e9-8806-ab238f7f964a.png)

Even though that Gravatar loaded just fine by itself:
https://2.gravatar.com/avatar/bf087bfd9864126accd9ce28335e26ef?s=150

It only happened on the first load though, so that's most likely safe to ignore.

jeherve committed:
Add rel noopener noreferrer

Co-Authored-By: eliorivero <eliorivero@gmail.com>

eliorivero committed:
Add final URL. Remove unused admin-head-XX action

eliorivero committed:
Copy wp_get_update_php_url, wp_get_default_update_php_url, and wp_update_php_annotation as methods into the class to avoid fatal errors

eliorivero commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,599 @@
+							/* translators: 1: "Update WordPress" screen URL, 2: "Update PHP" page URL */
+							' ' . wp_kses( __( '<a href="%1$s">Please update WordPress</a>, and then <a href="%2$s">learn more about updating PHP</a>.', 'jetpack' ), array( 'a' => array( 'href' => true ) ) ),
+							esc_url( self_admin_url( 'update-core.php' ) ),
+							esc_url( wp_get_update_php_url() )
```
Since they're a bit complex, I've decided to copy them as methods into the class.

eliorivero committed:
Use named arguments to avoid duplicates

eliorivero committed:
Make list of fallback avatars more diverse

eliorivero committed:
Merge branch 'master' into add/about-page

eliorivero commented:
Fixed conflict introduced by removal of `welcome-new-plan/style.scss` stylesheet.

crunnells committed:
Updated a8c stats to be less specific about # of employees, countries, and languages

crunnells committed:
Updated CSS so plugin images match core sizes, changed layout to flexbox instead of using floats, and tweaked some responsive behavior

crunnells committed:
Updated update_available sprintf() to use numbered placeholders

crunnells commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,599 @@
+						if ( $status['url'] ) {
+							if ( $compatible_php && $compatible_wp ) {
+								$action_links[] = sprintf(
+									'<a class="install-now button jptracks" data-slug="%s" href="%s" aria-label="%s" data-name="%s" data-jptracks-name="jetpack_about_install_button" data-jptracks-prop="%s">%s</a>',
```
Fixed in c4014a5

eliorivero committed:
Remove dynamic avatars

eliorivero committed:
Update link to install plugin based on whether the site is multisite. Path and query line are written into one line so it passes phpcs.

eliorivero commented:
Fixed issue with multisite. It's now working in multi and single sites. Note the network icon in the first two.

<img width="907" alt="Captura de Pantalla 2019-04-29 a la(s) 14 27 00" src="https://user-images.githubusercontent.com/1041600/56914998-d51da900-6a8b-11e9-9ccc-ae38636653b2.png">
<img width="982" alt="Captura de Pantalla 2019-04-29 a la(s) 14 27 43" src="https://user-images.githubusercontent.com/1041600/56914999-d51da900-6a8b-11e9-8ac1-b683a4ba362d.png">
<img width="970" alt="Captura de Pantalla 2019-04-29 a la(s) 14 29 08" src="https://user-images.githubusercontent.com/1041600/56915000-d5b63f80-6a8b-11e9-8bf1-3da1cad80321.png">

kraftbj commented on 3rd-party/woocommerce-services.php:
```diff
@@ -88,7 +88,7 @@ public function add_error_notice() {
 		?>
 		<div class="notice notice-error is-dismissible">
-			<p><?php _e( 'There was an error installing WooCommerce Services.', 'jetpack' ); ?></p>
+			<p><?php _e( 'This site is private.', 'jetpack' ); ?></p>
```
I think this was meant for another PR.

eliorivero replied:
Fixed

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+							</a>
+						</p>
+						<p>
+							<?php esc_html_e( 'We’re a distributed company with over 875 Automatticians in more than 67 countries speaking at least 83 different languages. Our common goal is to democratize publishing so that anyone with a story can tell it, regardless of income, gender, politics, language, or where they live in the world.', 'jetpack' ); ?>
```
For the next iteration, it would be nice if there was an endpoint we could hit that would save this value as a transient for something long (like a month?) so we don't have to worry about trying to update it whenever we hit 900, 925, 950, etc. Not blocking for this release.

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+			'woocommerce',
+			'wp-super-cache',
+			'wp-job-manager',
+			'co-authors-plus',
```
Likewise, would be good to make this something we can edit on our end. Non-blocking.

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+				<p class="jetpack-about__services-more">
+				<?php
+				echo wp_kses(
+					__( 'For even more of our WordPress plugins, please <a href="https://profiles.wordpress.org/automattic/#content-plugins" target="_blank" rel="noopener noreferrer" class="jptracks" data-jptracks-name="jetpack_about_wporg_profile">take a look at our WordPress.org profile</a>.', 'jetpack' ),
```
We should clean up the plugins in this profile. There's a lot of older ones that we don't maintain anymore that should probably be closed. Non-blocking.

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+
+					<div class="jetpack-about__text">
+						<p>
+							<?php esc_html_e( 'We are the people behind WordPress.com, WooCommerce, Jetpack, Simplenote, Longreads, VaultPress, Akismet, Gravatar, Crowdsignal, Cloudup, and more. We believe in making the web a better place.', 'jetpack' ); ?>
```
Do we want to highlight VaultPress, given the future branding goal?

Cloudup we're closed for new signups, so not sure if that is the best product to highlight either.

jeffgolenski replied:
I pulled this directly from the Automattic.com about page. I think it's fine to leave it for now.

kraftbj replied:
Good enough for me for now.

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+	 * {@see 'wp_update_php_url'} filter. Providing an empty string is not allowed and will result in the
+	 * default URL being used. Furthermore the page the URL links to should preferably be localized in the
+	 * site language.
+	 *
```
```suggestion
	 *
	 * @todo: Remove when 5.1 is minimum WP version.
```

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+	 * Do not use this function to retrieve this URL. Instead, use {@see wp_get_update_php_url()} when relying on the URL.
+	 * This function does not allow modifying the returned URL, and is only used to compare the actually used URL with the
+	 * default one.
+	 *
```
```suggestion
	 *
 	 * @todo: Remove when 5.1 is minimum WP version.
```

kraftbj commented on _inc/lib/admin-pages/class-jetpack-about-page.php:
```diff
@@ -0,0 +1,676 @@
+	 *
+	 * This function is to be used after {@see wp_get_update_php_url()} to display a consistent
+	 * annotation if the web host has altered the default "Update PHP" page URL.
+	 *
```
```suggestion
	 *
 	 * @todo: Remove when 5.1 is minimum WP version.
```

kraftbj reviewed (commented):
Left a few notes. @eliorivero can you say which, if any, you'd like to address. I'm okay if they're in another PR after this one is merged since functionality works (tested on single site and multisite domain installs), just minor-in-terms-of-code stuff.

kraftbj committed:
Add reminder to remove these functions

Adds a ToDo to the phpdoc of these functions copied from WP core so we remember to remove them when Jetpack supports only 5.1 and up.

Co-Authored-By: eliorivero <eliorivero@gmail.com>

eliorivero commented:
Thanks for the many throughout reviews @kraftbj They were super helpful in making this better.
Your points about serving that data that might change over time from and endpoint makes perfect sense, as well as the issue of cleaning up the plugin page. Can you please open new issues for those? Not sure who could work on the plugin page clean up, but you can assign the endpoint ones to me. I'm going to merge this now so it's ready for tomorrow. Thanks!

jeffgolenski commented:
Great work to all involved @crunnells @eliorivero @kraftbj @jeherve I appreciate you.
